src/openforms/payments/tests/test_views.py

```diff
@@ -0,0 +1,153 @@
+from decimal import Decimal
+from unittest.mock import patch
+from urllib.parse import quote
+
+from django.http import HttpResponseRedirect
+from django.test import TestCase, override_settings
+from django.test.client import RequestFactory
+
+from openforms.payments.models import SubmissionPayment
+from openforms.payments.tests.factories import SubmissionPaymentFactory
+from openforms.submissions.tests.factories import SubmissionFactory
+
+from ..base import BasePlugin, PaymentInfo
+from ..registry import Registry
+
+
+class Plugin(BasePlugin):
+    verbose_name = "some human readable label"
+    return_method = "GET"
+    webhook_method = "POST"
+
+    def start_payment(self, request, payment):
+        return PaymentInfo(type="get", url="http://testserver/foo")
+
+    def handle_return(self, request, payment):
+        return HttpResponseRedirect(payment.form_url)
+
+    def handle_webhook(self, request):
+        return None
+
+
+class ViewsTests(TestCase):
+    @override_settings(
+        CORS_ALLOW_ALL_ORIGINS=False, CORS_ALLOWED_ORIGINS=["http://foo.bar"]
+    )
+    @patch("openforms.payments.views.update_submission_payment_registration")
+    def test_views(self, update_payments_mock):
+        register = Registry()
+        register("plugin1")(Plugin)
+        plugin = register["plugin1"]
+        bad_plugin = Plugin("bad_plugin")
+
+        base_request = RequestFactory().get("/foo")
+
+        next_url_enc = quote("http://foo.bar")
+        bad_url_enc = quote("http://buzz.bazz")
+
+        registry_mock = patch("openforms.payments.views.register", new=register)
+        registry_mock.start()
+        self.addCleanup(registry_mock.stop)
+
+        submission = SubmissionFactory.create(
+            form__product__price=Decimal("11.25"),
+            form__payment_backend="plugin1",
+        )
+        self.assertTrue(submission.payment_required)
+
+        # check the start url
+        url = plugin.get_start_url(base_request, submission)
+        self.assertRegex(url, r"^http://")
```

Comment:
isn't it more readable to use `self.assertTrue(url.startswith("http://"))`?

Reply:
then the error message wouldn't list the problematic string